examples/serialize.py

```diff
@@ -0,0 +1,18 @@
+from transformers import AutoModelForCausalLM
+from tensorizer.serialization import TensorSerializer
+import torch
+
+model_ref = "EleutherAI/gpt-j-6B"
+# For less intensive requirements, swap above with the line below:
+# model_ref = "EleutherAI/gpt-neo-125m"
+model_name = model_ref.split("/")[-1]
+# Change this to your S3 bucket.
+s3_bucket = "bucket"
+s3_uri = f"s3://{s3_bucket}/{model_name}.tensors"
+
+model = AutoModelForCausalLM.from_pretrained(model_ref,
+                                             revision="float16",
+                                             torch_dtype=torch.float16)
+
+serializer = TensorSerializer(s3_uri)
+serializer.write_module(model)
```

Comment:
Without manually calling `close()`, running this example script causes an exception to be thrown due to `close()` being called automatically by the `__del__()` function. The ordering of gc messes it up as the imports are removed before the stream gets a chance to close.

@Eta0 - This might already be in one of your fixes but commenting here to make sure it's not forgotten about :) 

```suggestion
serializer.write_module(model)
serializer.close()
```

The Error:
```
✗  python serialize.py
Exception ignored in: <function TensorSerializer.__del__ at 0x7fe2821d0040>
Traceback (most recent call last):
  File "/home/npratt/git/tensorizer/tensorizer/serialization.py", line 790, in __del__
  File "/home/npratt/git/tensorizer/tensorizer/stream_io.py", line 403, in close
  File "/home/npratt/git/tensorizer/tensorizer/stream_io.py", line 237, in s3_upload
  File "/home/npratt/miniconda3/envs/tensorizer/lib/python3.8/site-packages/boto3/session.py", line 299, in client
  File "/home/npratt/miniconda3/envs/tensorizer/lib/python3.8/site-packages/botocore/session.py", line 976, in create_client
  File "/home/npratt/miniconda3/envs/tensorizer/lib/python3.8/site-packages/botocore/client.py", line 141, in create_client
  File "/home/npratt/miniconda3/envs/tensorizer/lib/python3.8/site-packages/botocore/client.py", line 189, in _create_client_class
  File "/home/npratt/miniconda3/envs/tensorizer/lib/python3.8/site-packages/botocore/client.py", line 503, in _create_methods
  File "/home/npratt/miniconda3/envs/tensorizer/lib/python3.8/site-packages/botocore/__init__.py", line 97, in xform_name
ModuleNotFoundError: import of re halted; None in sys.modules
```

Reply:
Fixed this behaviour in general in my branch on Friday night; it's now guaranteed to upload before interpreter shutdown. Going to add the `close()` call here too since it should have been there. Thanks.